cost/turbonomics/delete_unattached_volumes/aws/turbonomics_delete_virtual_volumes.pt

```diff
@@ -0,0 +1,364 @@
+
+##script using a manually acquired turbonomic cookie and pulls hard coded unused volume data
+script "js_get_turbonomics_recommendations", type: "javascript" do
+  result "request"
+  parameters "auth_cookie"
+  code <<-EOS
+  //replace cookie every day this is run
+    var request = {
+      verb: "POST",
+      pagination: "turbonomics_volumes_pagination_header",
+      host: "sales1.demo.turbonomic.com",
+      path: "/api/v3/markets/Market/actions",
+      body_fields: {
+        "actionStateList": ["READY", "ACCEPTED", "QUEUED", "IN_PROGRESS"],
+        "actionTypeList":["DELETE"],
+        "relatedEntityTypes":["VirtualVolume"],
+        "environmentType":"CLOUD",
+        "detailLevel":"EXECUTION",
+        "costType":"SAVING"
+      },
+      query_params: {
+        "limit": '100'
+      },
+      headers: {
+        "Content-Type": "application/json"
+        "Cookie": auth_cookie
+      }
+    }
+EOS
+end
+
+## verified that discovered is the only one we need
+script "js_get_business_units", type: "javascript" do
+  result "request"
+  parameters "auth_cookie", "param_provider"
+  code <<-EOS
+  //replace cookie every day this is run
+    var providers = {
+      "AWS":"AWS",
+      "Azure Subscription":"AZURE",
+      "GCP Project": "GCP",
+    }
+    var request = {
+      verb: "GET",
+      host: "sales1.demo.turbonomic.com",
+      path: "/api/v3/businessunits",
+      query_params: {
+        "cloud_type": providers[param_provider],
+        "type":"DISCOVERED"
+      },
+      headers: {
+        "Content-Type": "application/json"
+        "Cookie": auth_cookie
+      }
+    }
+EOS
+end
+
+script "js_filtered_turbonomics_recommendations", type: "javascript" do
+  result "result"
+  parameters "ds_get_turbonomics_recommendations", "param_days_unattached", "ds_get_business_units", "param_provider"
+  code <<-EOS
+    instances = []
+    monthlySavings = 0.0
+    function formatNumber(number, separator){
+    var numString =number.toString()
+    var values=numString.split(".")
+    var result = ''
+    while (values[0].length > 3){
+      var chunk = values[0].substr(-3)
+      values[0] = values[0].substr(0, values[0].length - 3)
+      result = separator + chunk + result
+    }
+    if (values[0].length > 0){
+      result = values[0] + result
+    }
+    if(values[1]==undefined){
+      return result
+    }
+    return result+"."+values[1]
+    }
+    _.each(ds_get_business_units, function(businessUnit) {
+      _.each(ds_get_turbonomics_recommendations, function(volume, index){
+        if (volume.businessUUID === businessUnit.uuid) {
+          ds_get_turbonomics_recommendations[index].accountID = businessUnit.accountID
+          ds_get_turbonomics_recommendations[index].accountName = businessUnit.displayName
+        }
+      })
+    })
+    var provider_service = {
+      "AWS":"EC2",
+      "Azure Subscription":"Microsoft.Compute",
+      "GCP Project": "Compute"
```

Comment:
I think this aren't the names for Virtual disk service, I think should be instead:
```
{
      "AWS": "EBS",
      "Azure Subscription": "Disk Storage",
      "GCP Project": "Persistent Disk"
}
```